### ISSUE TITLE
A question about the cursor

### ISSUE THREAD

**Issue by guuzaa** [pflenker/hecto-tutorial#16]
Hi @pflenker ,
 At first, thanks for your wonderful tutorial, that helps me study Rust progressively. 👍   Now, I'm learning chapter 3. However, I encountered a question I cannot figure out. As the following:
![image](https://user-images.githubusercontent.com/67539676/166137996-76140e31-5ec4-436a-a68c-cbad0cacfef2.png)
which is from :
 https://github.com/pflenker/hecto-tutorial/blob/9382faf2a780e23da462ed2c4b3907b75796e782/src/editor.rs#L62
Why are the positions of x and y in Line 70  different from those in Line 62? Could you share with me the reason?
Thanks again.

**Comment by xyjixyjixyji** [comment]
Hi, I have done following this tutorial today. In my implementation, the code is
```Rust
let Pos {mut x, mut y} = self.cursor_pos;
```
And it works fine with my code.

Therefore, you could try to code in {x, y} form to see whether it works fine or not, then you may modify to {y, x} to see what happens.

**Comment by pflenker** [comment]
Hey, 

thanks for your question!

Why this works: Since we use the same variable name in `move_cursor` as in the `Position` struct, the order of `mut y` and `mut x` in Line 62 does not matter.
For the same reason, it doesn't matter in line 70, the code there is just a shorthand for:
```
self.cursor_position = Position {
x: x,
y:y
}
```

which is just one of Rust's awesome features. 

Why did I do this? Beats me, I honestly can't remember. From the order in Line 62 and the order in which I deal with the keys in Line 64 and Line 65, I infer that I first added up and down movement before handling left and right movement. It seems when I cleaned up the code, I used the more intuitive order in Line 70 and missed the alternative order in Line 62.
